### PR TITLE
Add tests for Event. Remove redundant function 'def week' from Event.

### PR DESCRIPTION
**Pull request by differentreality**

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -141,10 +141,6 @@ def abstract_word_count
     abstract.to_s.split.size
   end
 
-  def week
```
Comment (ChrisBr) [inline app/models/event.rb]: ``week`` is not used?

File: spec/models/event_spec.rb [hunk 2]
```diff
@@ -1,8 +1,136 @@
...
+  let(:event) { create(:event) }
+  let(:user) { create(:user) }
 
-  describe 'abstract_word_count' do
+  describe 'validations' do
+    it 'has a valid factory' do
+      expect(build(:event)).to be_valid
+    end
+
+    it 'is not valid without a title' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Actually you're not testing what your test says and I'd prefer to use [shoulda-matcher](https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers) to test this.

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -1,8 +1,136 @@
...
+      expect(event.title.present?).to be true
+    end
+
+    it 'is not valid without an abstract' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: The same for this one!

File: spec/models/event_spec.rb [hunk 4]
```diff
@@ -1,8 +1,136 @@
...
+      expect(event.abstract.present?).to be true
+    end
+
+    it 'is not valid without a program' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: The same for this one!

File: spec/models/event_spec.rb [hunk 5]
```diff
@@ -1,8 +1,136 @@
...
+      expect(event.program.present?).to be true
+    end
+
+    it 'is not valid without an event_type' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: The same for this one!

File: spec/models/event_spec.rb [hunk 6]
```diff
@@ -1,8 +1,136 @@
...
+      expect(event.event_type.present?).to be true
+    end
+
+    context '#abstract_limit' do
+      before :each do
+        event.event_type.maximum_abstract_length = 2
+        event.event_type.minimum_abstract_length = 2
+      end
+
+      it 'returns error message if abstract is too long' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Maybe you can group this in a nested context ``valid`` and ``invalid``. That would make the test name shorter...

File: spec/models/event_spec.rb [hunk 7]
```diff
@@ -1,8 +1,136 @@
...
+      it 'does not return an error message if abstract length is within limits' do
+        event.abstract = 'Test abstract'
+        expect(event.valid?).to eq true
+        expect(event.errors.size).to eq 0
+      end
+    end
+
+    context '#before_end_of_conference' do
+      it 'adds error when event is created after the conference is over' do
+        conference = create(:conference, start_date: Date.today - 1, end_date: Date.today - 1)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: I'd also test the valid case

File: app/models/event.rb (hunk 1 above)
Comment (differentreality) [inline app/models/event.rb]: There is an attribute called week, which is set later on in the models/event.rb

File: spec/models/event_spec.rb (hunk 2 above)
Comment (ChrisBr) [inline spec/models/event_spec.rb]: What about doing it in one line?

```
it { is_expected.to validate_presence_of(:title) }
```

(same for the others)

Comment (ChrisBr) [inline spec/models/event_spec.rb]: Furthermore, the ``should`` syntax is deprecated!
http://betterspecs.org/#expect

File: spec/models/event_spec.rb [hunk 8]
```diff
@@ -1,8 +1,168 @@
...
 describe Event do
+  subject { create(:event) }
+  let(:conference) { create(:conference) }
+  let(:event) { create(:event, program: conference.program) }
+  let(:user) { create(:user) }
 
-  describe 'abstract_word_count' do
+  describe 'association' do
+    it 'should belong to program' do
+      should belong_to :program
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Please avoid using the deprecated ``should`` syntax
http://betterspecs.org/#expect

File: spec/models/event_spec.rb [hunk 9]
```diff
@@ -1,8 +1,168 @@
...
+    end
+
+    describe '#abstract_limit' do
+      before :each do
+        event.event_type.maximum_abstract_length = 2
+        event.event_type.minimum_abstract_length = 2
+      end
+
+      context 'is invalid' do
+
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Remove empty line please

File: spec/models/event_spec.rb [hunk 10]
```diff
@@ -1,8 +1,168 @@
...
+          event.abstract = 'Test abstract'
+          expect(event.valid?).to eq true
+          expect(event.errors.size).to eq 0
+        end
+      end
+    end
+
+    describe '#before_end_of_conference' do
+      context 'is invalid' do
+        before :each do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Not sure if it really necessary to have this in a before block if you only have one test, but no strong preference here.
(same for the next test)

File: spec/models/event_spec.rb [hunk 11]
```diff
@@ -1,8 +1,168 @@
...
+        end
+
+        it 'when event is created after the conference end_date, and returns an error message' do
+          expect(@new_event.valid?).to eq false
+          expect(@new_event.errors[:created_at]).to eq ["can't be after the conference end date!"]
+        end
+      end
+
+      context 'is valid' do
+        before :each do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Same here

File: spec/models/event_spec.rb [hunk 12]
```diff
@@ -1,8 +1,168 @@
...
+          expect(@new_event.valid?).to eq true
+        end
+      end
+    end
+  end
+
+  describe 'scope ' do
+    context 'confirmed' do
+      it 'returns only confirmed events' do
+        conference = create(:conference)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: You could move this and the next line in a before block.

File: spec/models/event_spec.rb [hunk 13]
```diff
@@ -40,4 +198,76 @@
...
+        new_event = create(:event)
+        speaker1 = create(:user, name: 'user speaker 1')
+        new_event.event_users = [create(:event_user, user: speaker1, event_role: 'submitter')]
+        new_event.event_users << [create(:event_user, user: speaker1, event_role: 'speaker')]
+
+        expect(new_event.speaker_names).to eq 'user speaker 1'
+      end
+
+      it 'when submitter is not a speaker' do
+        new_event = create(:event)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Please move the new_event in a let(:new_user) block

File: spec/models/event_spec.rb [hunk 14]
```diff
@@ -1,8 +1,152 @@
...
 describe Event do
+  subject { create(:event) }
+  let(:conference) { create(:conference) }
+  let(:event) { create(:event, program: conference.program) }
+  let(:new_event) { create(:event) }
+  let(:user) { create(:user) }
 
-  describe 'abstract_word_count' do
+  describe 'association' do
+    it 'should belong to program' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Please also use one line expectations here

File: spec/models/event_spec.rb [hunk 15]
```diff
@@ -1,8 +1,152 @@
...
+      is_expected.to belong_to :program
+    end
+
+    it 'should belong to event_type' do
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Please also use one line expectations here

File: spec/models/event_spec.rb [hunk 16]
```diff
@@ -1,8 +1,152 @@
...
+          expect(new_event.valid?).to eq true
+        end
+      end
+    end
+  end
+
+  describe 'scope ' do
+    context 'confirmed' do
+      before :each do
+        conference = create(:conference)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: why not let()?

File: spec/models/event_spec.rb [hunk 17]
```diff
@@ -1,8 +1,152 @@
...
+        create(:event, program: conference.program)
+      end
+
+      it 'returns only confirmed events' do
+        my_event = create(:event, state: 'confirmed', program: conference.program)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: and when you use conference here, you use the conference create in the let in line 5 and not the conference created in the before block! So either remove line 82 if you don't need it or move it into a let or use an instance variable.

File: spec/models/event_spec.rb [hunk 18]
```diff
@@ -1,8 +1,152 @@
...
+
+        expect(conference.program.events.confirmed).to eq [my_event]
+      end
+    end
+
+    context 'highlighted' do
+      it 'returns only highlighted events' do
+        my_event = create(:event, is_highlight: true, program: conference.program)
```
Comment (ChrisBr) [inline spec/models/event_spec.rb]: Same here!

**Comment by hennevogel** [comment]
Is there anything left to do? @ChrisBr

**Comment by ChrisBr** [comment]
:+1:

**Pull request merged**